benchopt/runner.py

```diff
@@ -90,17 +90,10 @@ def run_one_to_cvg(benchmark, objective, solver, meta, stopping_criterion,
 
     curve = []
     with exception_handler(output, pdb=pdb) as ctx:
+        output.progress('empty run for compilation')
+        solver.warmup_solver()
```

Comment:
should we warmup all solvers? I'm thinking of solvers which are very slow even for one iteration (need to compute an SVD for example), this will double their running time

Reply:
I don't really know what is the best option. Note that I merely harmonized: we used to do it for all `callback` solver, but not for the others...

I like that it is automatic and easy to override by implementing the method.
But I agree it might be costly.

Reply:
We should decide this point soon. maybe tqlking together?

Reply:
Actually, warmup does not make sense for `SingleRunCriterion` for instance...
The default should probably depends on many things... I think trying to be smart will shoot us in the foot.

IMO, the solvers should be responsible to make the pre-compilation call in `set_objective` instead of using `warmup` function. To make it easy, we could provide a `run_once` function that take the solver and run it once but the user need to explicitely call it? WDYT?